master/buildbot/reporters/gerrit.py

```diff
@@ -329,7 +329,8 @@ def getBuildInfo(build):
                         'result': result,
                         'resultText': resultText,
                         'text': build['state_string'],
-                        'url': utils.getURLForBuild(self.master, build['builder']['builderid'], build['number'])
+                        'url': utils.getURLForBuild(self.master, build['builder']['builderid'], build['number']),
```

Comment:
build.getUrl is from process.build.Build class, which is the running build object which you can get in your custom steps.

in reporters, you are bound to the data api.